File: python practice/20260615_Height_of_BST.py
import sys

class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def insert(root, val):
    if not root:
        return Node(val)
    if val < root.val:
        root.left = insert(root.left, val)
    else:
        root.right = insert(root.right, val)
    return root

def height(root):
    if not root:
        return 0
    return 1 + max(height(root.left), height(root.right))

def solve():
    data = sys.stdin.read().split()
    idx = 0
    while idx < len(data):
        n = int(data[idx])
        idx += 1
        root = None
        for _ in range(n):
            root = insert(root, int(data[idx]))
            idx += 1
        print(height(root))

if __name__ == '__main__':
    solve()

File: C Language/20260615_Height_of_BST.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int val;
    struct Node *left, *right;
} Node;

Node* insert(Node* root, int val) {
    if (!root) {
        Node* node = (Node*)malloc(sizeof(Node));
        node->val = val;
        node->left = node->right = NULL;
        return node;
    }
    if (val < root->val) {
        root->left = insert(root->left, val);
    } else {
        root->right = insert(root->right, val);
    }
    return root;
}

int height(Node* root) {
    if (!root) return 0;
    int left_h = height(root->left);
    int right_h = height(root->right);
    return 1 + (left_h > right_h ? left_h : right_h);
}

int main() {
    int n;
    while (scanf("%d", &n) == 1) {
        Node* root = NULL;
        for (int i = 0; i < n; i++) {
            int val;
            scanf("%d", &val);
            root = insert(root, val);
        }
        printf("%d\n", height(root));
    }
    return 0;
}